### ISSUE TITLE
wlr_output_get_primary_formats returning NULL not handled (causing segfault)

### ISSUE THREAD

**Issue by sertonix** [swaywm/sway#8496]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions on IRC at #sway on Libera Chat.
- Proprietary graphics drivers, including nvidia, are not supported. Please use the open source equivalents, such as nouveau, if you would like to use Sway.
- Please do NOT submit issues for information from the github wiki. The github wiki is community maintained and therefore may contain outdated information, scripts that don't work or obsolete workarounds.
  If you fix a script or find outdated information, don't hesitate to adjust the wiki page.

### Please fill out the following:
- **Sway Version: 1.10**

- **Debug Log: [sway.log](https://github.com/user-attachments/files/18083880/sway.log)**

- **Configuration File: `xwayland disable`**

- **Stack Trace: [sway-bt.log](https://github.com/user-attachments/files/18084090/sway-bt.log)**

- **Description:**

If the wlroots documentation is correct `wlr_output_get_primary_formats` may return `NULL` ([doc](https://wayland.emersion.fr/wlroots/wlr/types/wlr_output.h.html#func-wlr_output_get_primary_formats)). It looks like this is not handled correctly at https://github.com/swaywm/sway/blob/f293418d9d8a9e71fc52bc75a83d24ec2cc934c9/sway/config/output.c#L776 which passes the return value to [`wlr_drm_format_set_get`](https://github.com/swaywm/sway/blob/f293418d9d8a9e71fc52bc75a83d24ec2cc934c9/sway/config/output.c#L786) without a `NULL` check. wlroots then tries to dereference the pointer and segfaults ([`drm_format_set.c#L32`](https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/9649fbe443a96680f47e550ebaf37bba0fe44c9e/render/drm_format_set.c#L32)).

I wasn't able to create a reproducer that I can share but I hope the analysis is enough.

**Comment by emersion** [comment]
Indeed! We should add a NULL check right before `wlr_drm_format_set_get()`. Would you be willing to send a patch?

**Comment by sertonix** [comment]
I don't know in which line the NULL check belongs and what should be done in that case. So I would prefer to leave that to the experts. Hope that is ok.
